File: javascript/binary/binary.js
var Binary = function (binary) {
  this.binary = binary;
};

Binary.prototype.toDecimal = function () {
  var result = 0;
  for (var i = 0; i < this.binary.length; i++) {
    if (this.binary[i] == '0' || this.binary[i] == '1') {
      result += this.binary[i] * Math.pow(2, this.binary.length - (i + 1));
    } else {
      return 0;
    }
  }
  return result;
};

module.exports = Binary;

File: csharp/binary/Binary.cs
using System;

public class Binary
{
  public static int ToDecimal(string binary)
  {
    var result = 0;
    for (var i = 0; i < binary.Length; i++)
    {
      if (char.IsLetter(binary[i]))
        return 0;
      var value = char.GetNumericValue(binary[i]);
      if (value < 0 || value > 1)
        return 0;
      result += (int)(value * Math.Pow(2.0, binary.Length - (i + 1)));
    }
    return result;
  }
}